build/android/native/jni/Android.mk

```diff
@@ -0,0 +1,249 @@
+LOCAL_PATH := $(call my-dir)/..
+
+#LOCAL_ADDRESS_SANITIZER:=true
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := Curl
+LOCAL_SRC_FILES := deps/Android/Curl/${COMPILER_VERSION}/$(APP_ABI)/libcurl.a
+include $(PREBUILT_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := Freetype
+LOCAL_SRC_FILES := deps/Android/Freetype/${COMPILER_VERSION}/$(APP_ABI)/libfreetype.a
+include $(PREBUILT_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := Irrlicht
+LOCAL_SRC_FILES := deps/Android/Irrlicht/${COMPILER_VERSION}/$(APP_ABI)/libIrrlicht.a
+include $(PREBUILT_STATIC_LIBRARY)
+
+#include $(CLEAR_VARS)
+#LOCAL_MODULE := LevelDB
+#LOCAL_SRC_FILES := deps/Android/LevelDB/${COMPILER_VERSION}/$(APP_ABI)/libleveldb.a
+#include $(PREBUILT_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := OpenAL
+LOCAL_SRC_FILES := deps/Android/OpenAL-Soft/${COMPILER_VERSION}/$(APP_ABI)/libopenal.a
+include $(PREBUILT_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := Vorbis
+LOCAL_SRC_FILES := deps/Android/Vorbis/${COMPILER_VERSION}/$(APP_ABI)/libvorbis.a
+include $(PREBUILT_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := Minetest
+
+LOCAL_CFLAGS += \
+	-DJSONCPP_NO_LOCALE_SUPPORT      \
+	-DHAVE_TOUCHSCREENGUI            \
+	-DENABLE_GLES=1                  \
+	-DUSE_CURL=1                     \
+	-DUSE_SOUND=1                    \
+	-DUSE_FREETYPE=1                 \
+	-DUSE_LEVELDB=0                  \
+	-DVERSION_MAJOR=${versionMajor}  \
+	-DVERSION_MINOR=${versionMinor}  \
+	-DVERSION_PATCH=${versionPatch}  \
+	$(GPROF_DEF)
+
+ifdef NDEBUG
+	LOCAL_CFLAGS += -DNDEBUG=1
+endif
+
+ifdef GPROF
+	GPROF_DEF := -DGPROF
+	PROFILER_LIBS := android-ndk-profiler
+	LOCAL_CFLAGS += -pg
+endif
+
+LOCAL_C_INCLUDES := \
+	../../../src                                   \
+	../../../src/script                            \
+	../../../lib/gmp                               \
+	../../../lib/lua/src                           \
+	../../../lib/jsoncpp                           \
+	deps/Android/Curl/include                      \
+	deps/Android/Freetype/include                  \
+	deps/Android/Irrlicht/include                  \
+	deps/Android/LevelDB/include                   \
+	deps/Android/libiconv/include                  \
+	deps/Android/libiconv/libcharset/include       \
+	deps/Android/OpenAL-Soft/include               \
+	deps/Android/sqlite                            \
+	deps/Android/Vorbis/include
+
+LOCAL_SRC_FILES := \
+	../../../src/ban.cpp                           \
+	../../../src/chat.cpp                          \
+	../../../src/clientiface.cpp                   \
+	../../../src/collision.cpp                     \
+	../../../src/content/content.cpp               \
```

Comment:
`content/, `mapgen/`, and `daabase/` should be moved to the globs below this, unless there's a particular reason why you do this